### ISSUE TITLE
Console error : Uncaught TypeError: Cannot read properties of undefined (reading 'length')

### ISSUE THREAD

**Issue by Leeroy-agency** [weareferal/craft-matrix-field-preview#88]
We have a project with your plugin and https://plugins.craftcms.com/neo and there is an error in the browser console, when we are on a page with Neo field.

It append when the script BlockTypeModal.js try to create a new instance :

`Uncaught TypeError: Cannot read properties of undefined (reading 'length')`
BlockTypeModal.js line 285

![mfp-js-bug](https://user-images.githubusercontent.com/65175280/178557755-6751c615-625f-4089-8eaa-edcc6f4768a6.jpg)
![mfp-js-bug2](https://user-images.githubusercontent.com/65175280/178557813-2ae12630-a4c7-4952-8c0b-b27efcb4b4c9.jpg)

The thing is that the init method of BlockTypeModal is called twice, once with an Object with the categories attribute and second time with this object :
```
{
    "iconTypeBlock": {
        "name": "Icon Type Block",
        "handle": "iconTypeBlock",
        "description": "",
        "descriptionHTML": "",
        "categoryId": 3,
        "image": null,
        "thumb": null
    }
}
```

This block is a Neo block that can't be put at the top level of the content.

This bug does not seem to break anything, but it's a bit annoying :slightly_smiling_face: and it brinks another problem with the Neo integration.

With Neo, we can create block that are not at "Top level" : that are child to another blocks. In our opinion, theses blocks should not be displayed in the in the Matrix Field Preview modal.

For now, we put all block that we want to be hidden into a category that we hide with a js script. Maybe a new feature with a way to exclude some blocks in the preview could be awesome!

**Comment by timmyomahony** [comment]
Thanks for the report 👍🏻 I'll look into it

**Comment by robinsm** [comment]
Hey @timmyomahony .  Was just about to report this issue and can see it's already reported in great detail.  We're experiencing the same issues as documented here.

**Comment by timmyomahony** [comment]
Apologies @Leeroy-agency for the late reply, I'm catching up on these Neo issues now.

Just to confirm, this issue was happening on page-load, when viewing an entry that contained your Neo field? And your Neo field is configured with both top-level and non-top-level blocks?

**Comment by Leeroy-agency** [comment]
Exact

> Apologies @Leeroy-agency for the late reply, I'm catching up on these Neo issues now.
> 
> Just to confirm, this issue was happening on page-load, when viewing an entry that contained your Neo field? And your Neo field is configured with both top-level and non-top-level blocks?

Exact.

**Comment by timmyomahony** [comment]
Hi @Leeroy-agency this should be fixed in the latest release 4.0.5

https://github.com/weareferal/craft-matrix-field-preview/releases/tag/4.0.5

There were a number of Neo issues that I've tried to tidy up so Neo performance should be much better now. Let me know if it fixes the errors you were seeing.

**Comment by Leeroy-agency** [comment]
Thanks you!

That's fixed it!
